guilib/measure/classes/abc.py

```diff
@@ -0,0 +1,603 @@
+            Larger values in the tuple indicate a higher degree of
+            conformity. The order of the items in the tuple is the
+            order of their significance. This return value is used
+            to determine the best-matching settings files when
+            multiple files are found.
+        """
+        if not isinstance(obj_info, SettingsInfo):
+            raise TypeError(
+                f'obj_info should be of type SettingsInfo '
+                f'but is of type {type(obj_info).__name__}.'
+                )
+        return tuple()
+
+    @classmethod
+    @abstractmethod
+    def is_settings_for_this_class(cls, config):
+        """Determine if a `config` file is for this class.
+
+        This method must use cls attributes to determine if
+        a config is suitable for the class.
+
+        Parameters
+        ----------
+        config : ConfigParser
+            The settings to check.
+
+        Returns
+        -------
+        is_suitable : bool
+            True if the settings file is for the class.
+        """
+
+    def are_settings_invalid(self, new_settings):
+        """Check if there are any invalid settings.
+
+        Reimplementations can add additional mandatory settings at
+        runtime. The base implementation will check if all of the
+        _mandatory_settings are present in the provided settings.
+
+        Parameters
+        ----------
+        new_settings : ViPErLEEDSettings
+            The new settings.
+
+        Returns
+        -------
+        invalid_settings : list of tuples
+            Invalid _mandatory_settings of self as a list of tuples.
+            The first entry in each tuple can be either '<section>',
+            '<section>/<option>', or
+            '<section>/<option> not one of <value1>, <value2>, ...'.
+            Further entries, added by reimplementations, are
+            information on what is wrong with the settings.
+        """
+        return [(invalid,) for invalid in
+                new_settings.has_settings(*self._mandatory_settings)]
+
+    @abstractmethod
+    def get_settings_handler(self):
+        """Return a SettingsHandler object for displaying settings.
+
+        This method should be extended in subclasses, i.e., do
+        handler = super().get_settings_handler(), and then add
+        appropriate sections and/or options to it using the
+        handler.add_section, and handler.add_option methods.
+
+        The base-class implementation returns a handler that
+        contains the location of the settings file.
+
+        Returns
+        -------
+        handler : SettingsHandler
+            The handler used in a SettingsDialog to display the
+            settings of this instance to users.
+        """
+        self.check_creating_settings_handler_is_possible()
+        handler = SettingsHandler(self.settings)
+        return handler
+
+    def set_settings(self, new_settings):
```

Comment:
Do we want to make this a `pyqtSlot`? If yes, make sure to propagate decoration to subclasses that extend this.